qcodes/instrument_drivers/american_magnetics/AMI430.py

```diff
@@ -472,8 +474,10 @@ def _update_units(self, ramp_rate_units=None, field_units=None):
 
 
 class AMI430_3D(Instrument):
-    def __init__(self, name, instrument_x, instrument_y,
-                 instrument_z, field_limit, **kwargs):
+    def __init__(self, name,
+                 instrument_x, instrument_y, instrument_z,
+                 field_limit: Union[numbers.Real, Iterable[Callable]],
```

Comment:
Should this be a `Callable[[numerictype]], bool]` to make it even more strict

Reply:
sure. shall i use numbers.Real again for `numerictype`?

Reply:
Yes if that works. Otherwise we may need to define a union in types.py

Reply:
done.